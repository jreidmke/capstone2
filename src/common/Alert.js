import Alert from 'react-bootstrap/Alert';
import {useState} from 'react';

const CustomAlert = ({color, message}) => {
    const [isShowing, setIsShowing] = useState(true); 

    if(isShowing) {
        return(
            <Alert variant={color} dismissible onClose={() => setIsShowing(false)}><b>{message}</b></Alert>
            )
    };
        return(
            <div></div>
        )
    };

export default CustomAlert;

