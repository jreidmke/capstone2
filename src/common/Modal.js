import {useState} from 'react';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import Modal from 'react-bootstrap/Modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faQuestion, faQuestionCircle } from '@fortawesome/free-solid-svg-icons'

  function CustomModal({message}) {
    const [show, setShow] = useState(false);
  
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
  
    return (
      <>

        <a><FontAwesomeIcon icon={faQuestionCircle} onClick={handleShow} className='ml-3'/></a>
  
        <Modal show={show} onHide={handleClose}>
          <Modal.Header closeButton>
            <Modal.Title>{message.heading}</Modal.Title>
          </Modal.Header>
          <Modal.Body>{message.body}</Modal.Body>
          <Modal.Footer>
            <Button variant="secondary" onClick={handleClose}>
              Close
            </Button>
          </Modal.Footer>
        </Modal>
      </>
    );
  }

export default CustomModal; 