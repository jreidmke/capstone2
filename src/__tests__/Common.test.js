import React from "react";
import { render } from "@testing-library/react";
import Alert from './../common/Alert';
import LoadingSpinner from './../common/LoadingSpinner';
import Modal from './../common/Modal';

test('renders without crashing', function() {
    render(<Alert color='success' message='Lorem ipsum'/>); 
})

test('it matches snapshot', function() {
    const {asFragment} = render(
        <Alert color='success' message='Lorem ipsum'/>
    );
    expect(asFragment()).toMatchSnapshot();
}); 

test('renders without crashing', function() {
    render(<LoadingSpinner/>); 
})

test('it matches snapshot', function() {
    const {asFragment} = render(
        <LoadingSpinner/>
    );
    expect(asFragment()).toMatchSnapshot();
}); 

test('renders without crashing', function() {
    render(<Modal message='Lorem ipsum'/>); 
})

test('it matches snapshot', function() {
    const {asFragment} = render(
        <Modal message='Lorem ipsum'/>
    );
    expect(asFragment()).toMatchSnapshot();
}); 