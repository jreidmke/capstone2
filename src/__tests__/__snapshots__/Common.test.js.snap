// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success alert-dismissible show"
    role="alert"
  >
    <button
      class="close"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
      <span
        class="sr-only"
      >
        Close alert
      </span>
    </button>
    <b>
      Lorem ipsum
    </b>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot 3`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success alert-dismissible show"
    role="alert"
  >
    <button
      class="close"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
      <span
        class="sr-only"
      >
        Close alert
      </span>
    </button>
    <b>
      Lorem ipsum
    </b>
  </div>
</DocumentFragment>
`;

exports[`renders without crashing and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
